### DISCUSSION TITLE
Running webserver custom port on server accessible only to clients

### DISCUSSION THREAD

**Discussion by MoHasanJ** [strongswan/strongswan#1828] [Q&A]
Hello,

In this setup, how can I achieve that I have a webserver running on port :2312 and I want it to be accessible to clients connected to VPN server. 
I tried configuring server on port :2312 and I can access it locally and can access it publicly if allowed in firewall but I can't connect to webserver at port 2312 from connected VPN client. 

So want this : 
webserver on port 2312 running on VPN host, any client  that is connected to this VPN host can access that server on port 2312, but I can't achieve this as I don't know which IP to be used.

**Comment by tobiasbrunner** [comment]
> but I can't achieve this as I don't know which IP to be used

What do you mean? What IP?

Anyway, you might want to look into the [`policy` match](https://ipset.netfilter.org/iptables-extensions.man.html) for iptables. Or you block all traffic and use the [default updown script](https://docs.strongswan.org/docs/5.9/plugins/updown.html) to install client-specific rules dynamically (with `hostaccess` enabled so it adds them to INPUT/OUTPUT and not just FORWARD).

**Reply by MoHasanJ** [reply]
May be I was not clear. 
I have strongswanvpn configured using in roadwarrior method ie. multiple clients connect to this VPN hosts / server. What I want is to run a webserver on this VPN host and that webserver should be accessible only to clients connected. So I configured webserver on port `:2312` . I had assumption that I could use some local address (example I am using `192.168.0.1/24 ` can assigne like `192.168.0.254` vpn host and then any client connected to it can access that webserver using "192.168.0.254:2312" but is is not happening as alias IP (192.168.0.254) assigned to VPN host is not reachable from clients computer so can't access this webserver running on VPN host from clients. Any idea how can I achieve that ?

**Reply by tobiasbrunner** [reply]
Sure, that should just work. If not, you configured something incorrectly (config, address, firewall etc.).

**Reply by MoHasanJ** [reply]
Thanks, I figured is firewall, however I am not sure what is blocking, here is the iptables output: (this firewall was already there)

Local IP range is : `10.10.0.0/16`
Added alias IP as `10.10.0.254/16` to VPN node so that can access webserver as : `10.10.0.254:2312`

Tried to whitelist port `2312` but no success. 

 ```
Generated by iptables-save v1.8.7 on Fri Aug  4 21:34:36 2023
*mangle
:PREROUTING ACCEPT [36049:13749088]
:INPUT ACCEPT [14525:2490674]
:FORWARD ACCEPT [20671:9354299]
:OUTPUT ACCEPT [20393:12303483]
:POSTROUTING ACCEPT [39032:21357089]
-A FORWARD -s 10.10.0.0/16 -o ens3 -p tcp -m policy --dir in --pol ipsec -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
COMMIT
# Completed on Fri Aug  4 21:34:36 2023
# Generated by iptables-save v1.8.7 on Fri Aug  4 21:34:36 2023
*filter
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [18361:12002790]
-A INPUT -j LOG
-A INPUT -i ens3 -m state --state NEW -m recent --update --seconds 300 --hitcount 60 --name DEFAULT --mask 255.255.255.255 --rsource -j DROP
-A INPUT -i ens3 -m state --state NEW -m recent --set --name DEFAULT --mask 255.255.255.255 --rsource
-A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT
-A INPUT -i lo -j ACCEPT
-A INPUT -m state --state INVALID -j DROP

-A INPUT -p tcp -m tcp --dport 22 -j ACCEPT
-A INPUT -p udp -m udp --dport 500 -j ACCEPT
-A INPUT -p udp -m udp --dport 4500 -j ACCEPT

#Allowed port 2312 for custom webserver
-A INPUT  -s 10.10.0.0/16 -p tcp --dport 2312 -j ACCEPT

-A INPUT -j DROP
-A FORWARD -s 10.10.0.0/16 -m policy --dir in --pol ipsec --proto esp -j ACCEPT
-A FORWARD -d 10.10.0.0/16 -m policy --dir out --pol ipsec --proto esp -j ACCEPT
-A FORWARD -j DROP
COMMIT
# Completed on Fri Aug  4 21:34:36 2023
# Generated by iptables-save v1.8.7 on Fri Aug  4 21:34:36 2023
*nat
:PREROUTING ACCEPT [834:86522]
:INPUT ACCEPT [7:364]
:OUTPUT ACCEPT [454:38699]
:POSTROUTING ACCEPT [454:38699]
-A POSTROUTING -s 10.10.0.0/16 -o ens3 -m policy --dir out --pol ipsec -j ACCEPT
-A POSTROUTING -s 10.10.0.0/16 -o ens3 -j MASQUERADE
COMMIT
# Completed on Fri Aug  4 21:34:36 2023
```
With my limited knowledge I can not figure which is exactly blocking port 2312 from working. As soon as I flush firewall, I can connect to webserver on `10.10.0.254:2312`
